scripts/lint/lint.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
```

Comment:
I would say overall i find this script a bit hard to read but get the gist, bit hard to tell what the text manipulation is doing etc but probably fine since this is best effort

Reply:
Anything in particular I could do to to help? Add more comments, an quick primer on Ninja syntax, logging?